superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricEditPopover.jsx

```diff
@@ -110,20 +107,26 @@ export default class AdhocMetricEditPopover extends React.Component {
     this.props.getCurrentTab(this.defaultActiveTabKey);
   }
 
+  componentDidUpdate(prevProps, prevState) {
+    if (
+      prevState.adhocMetric?.label !== this.state.adhocMetric?.label ||
+      prevState.savedMetric?.metric_name !== this.state.savedMetric?.metric_name
```

Comment:
do we need to add a `prevState.savedMetric?.verbose_name !== this.state.savedMetric?.verbose_name` here?

Reply:
I don't think so - `metric_name` check is used only to determine whether a different saved metric has been chosen. `metric_name` is unique, so it's sufficient to compare only this field.